flink-runtime/src/main/java/org/apache/flink/runtime/operators/ReduceCombineDriver.java

```diff
@@ -42,34 +44,38 @@
  * Combine operator for Reduce functions, standalone (not chained).
  * Sorts and groups and reduces data, but never spills the sort. May produce multiple
  * partially aggregated groups.
- * 
+ *
  * @param <T> The data type consumed and produced by the combiner.
  */
 public class ReduceCombineDriver<T> implements Driver<ReduceFunction<T>, T> {
-	
+
 	private static final Logger LOG = LoggerFactory.getLogger(ReduceCombineDriver.class);
 
 	/** Fix length records with a length below this threshold will be in-place sorted, if possible. */
 	private static final int THRESHOLD_FOR_IN_PLACE_SORTING = 32;
-	
-	
+
+
 	private TaskContext<ReduceFunction<T>, T> taskContext;
 
 	private TypeSerializer<T> serializer;
 
 	private TypeComparator<T> comparator;
-	
+
 	private ReduceFunction<T> reducer;
-	
+
 	private Collector<T> output;
-	
+
+	private DriverStrategy strategy;
+
 	private InMemorySorter<T> sorter;
-	
+
 	private QuickSort sortAlgo = new QuickSort();
 
+	private ReduceHashTable<T> table;
+
 	private List<MemorySegment> memory;
 
-	private boolean running;
+	private volatile boolean canceled;
```

Comment:
I have now looked more into this `volatile` stuff, and the problem is that further investigation would be needed to ensure that the variable being `volatile` doesn't have a bad performance effect (see [1]). So maybe the best option for now is to make it non-volatile (as it is in all the other drivers) and just hope that it doesn't blow up. Should I open a Jira for doing further investigation and then maybe changing all the drivers?

[1] http://brooker.co.za/blog/2012/09/10/volatile.html